### ISSUE TITLE
Rule Change: Rename required option in fenced-code-language

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#475]
### What rule do you want to change?

fenced-code-language

### What change do you want to make?

Rename an option

### How do you think the change should be implemented?

Other

### Example code

\

\```js
const message = "Hello, world!";
console.log(message);
\```

### What does the rule currently do for this code?

The `required` option specifies a list of languages that must be used for fenced code blocks. If a code block uses a language not in this list, it's flagged as an error.

### What will the rule do after it's changed?

The rule's functionality will remain the same. The only change will be the name of the option from `required` to `allowLanguages`. This new name will more accurately reflect that the option defines a whitelist of permissible languages, reducing the potential confusion that `required` might imply a boolean value.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

[MD040](https://github.com/DavidAnson/markdownlint/blob/main/doc/md040.md) also uses a similar naming convention with `allowed_languages`.
